wit/stream-producer.wit

```diff
@@ -0,0 +1,6 @@
+default interface producer { 
+    use pkg.types.{message, error, sender}
+
+    send-message: func(s: sender, m: message, partition: u32, offset: u32) -> result<_, error>
+    send-message-batch: func(s: sender, m: list<message>, partition: u32, offset: u32) -> result<_, error>
```

Comment:
The producer interface should be the same for queues and streams. stash the partition info into the factory. you also don't send to offsets. that is a consumer thing.